crates/libs/result/src/error.rs

```diff
@@ -1,10 +1,58 @@
 use super::*;
+use crate::HRESULT;
 use core::ffi::c_void;
 
+#[allow(dead_code)] // for cases that are not yet stabilized; see below
+mod winerror {
+    use crate::HRESULT;
+
+    // We define these here for non-Windows platforms.
+    // It's just as easy to use them on Windows, too.
+
+    // Keep sorted by numeric value
+    pub const E_UNEXPECTED: HRESULT = HRESULT::from_u32(0x8000FFFF);
+    pub const E_FAIL: HRESULT = HRESULT::from_u32(0x80004005);
+    pub const E_OUTOFMEMORY: HRESULT = HRESULT::from_u32(0x8007000E);
+    pub const E_INVALIDARG: HRESULT = HRESULT::from_u32(0x80070057);
+
+    // Keep sorted by numeric value
+    pub const ERROR_ACCESS_DENIED: u32 = 5;
+    pub const ERROR_INVALID_DATA: u32 = 13;
+    pub const ERROR_HANDLE_EOF: u32 = 38;
+    pub const ERROR_HANDLE_DISK_FULL: u32 = 39;
+    pub const ERROR_NOT_SUPPORTED: u32 = 50;
+    pub const ERROR_BROKEN_PIPE: u32 = 109;
+    pub const ERROR_SEEK_ON_DEVICE: u32 = 132;
+    pub const ERROR_DIR_NOT_EMPTY: u32 = 145;
+    pub const ERROR_ALREADY_EXISTS: u32 = 183;
+    pub const ERROR_DIRECTORY_NOT_SUPPORTED: u32 = 336;
+    pub const ERROR_NO_UNICODE_TRANSLATION: u32 = 1113;
+    pub const ERROR_POSSIBLE_DEADLOCK: u32 = 1131;
+    pub const ERROR_NOT_FOUND: u32 = 1168;
+    pub const ERROR_DISK_QUOTA_EXCEEDED: u32 = 1295;
+    pub const ERROR_TIMEOUT: u32 = 1460;
+    pub const ERROR_FILE_READ_ONLY: u32 = 6009;
+    pub const WSAEINTR: u32 = 10004;
+    pub const WSAEWOULDBLOCK: u32 = 10035;
+    pub const WSAEADDRINUSE: u32 = 10048;
+    pub const WSAEADDRNOTAVAIL: u32 = 10049;
+    pub const WSAENETDOWN: u32 = 10050;
+    pub const WSAENETUNREACH: u32 = 10051;
+    pub const WSAECONNABORTED: u32 = 10053;
+    pub const WSAECONNRESET: u32 = 10054;
+    pub const WSAENOTCONN: u32 = 10057;
+    pub const WSAECONNREFUSED: u32 = 10061;
+    pub const WSAEHOSTUNREACH: u32 = 10065;
```

Comment:
This interop with `ErrorKind` seems like it could very well be a separate discussion and potential feature. I'm not convinced it's really worth baking in to the `windows-result` crate but I am confident it is not essential to non-Windows support for COM.

Reply:
Agreed.  The existing conversion is broken, though.

Reply:
Reverted.  I might circle back to the broken conversion later.